lib/msf/ui/console/module_argument_parsing.rb

```diff
@@ -0,0 +1,200 @@
+  ))
+
+  @@exploit_opts = Rex::Parser::Arguments.new(@@module_opts.fmt.merge(
+    '-e' => [ true,  'The payload encoder to use.  If none is specified, ENCODER is used.' ],
+    '-f' => [ false, 'Force the exploit to run regardless of the value of MinimumRank.'    ],
+    '-n' => [ true,  'The NOP generator to use.  If none is specified, NOP is used.'       ],
+    '-p' => [ true,  'The payload to use.  If none is specified, PAYLOAD is used.'         ],
+    '-t' => [ true,  'The target index to use.  If none is specified, TARGET is used.'     ],
+    '-z' => [ false, 'Do not interact with the session after successful exploitation.'     ]
+  ))
+
+  def parse_check_opts(args)
+    help_cmd = proc do |_result|
+      cmd_check_help
+    end
+    parse_opts(@@module_opts_with_action_support, args, help_cmd: help_cmd)&.slice(:datastore_options)
+  end
+
+  def parse_run_opts(args, action: nil)
+    help_cmd = proc do |result|
+      if result[:action].nil?
+        cmd_run_help
+      else
+        cmd_action_help(action)
+      end
+    end
+
+    parse_opts(@@module_opts_with_action_support, args, help_cmd: help_cmd)
+  end
+
+  def parse_exploit_opts(args)
+    help_cmd = proc do |_result|
+      cmd_exploit_help
+    end
+    parse_opts(@@exploit_opts, args, help_cmd: help_cmd)&.except(:action)
+  end
+
+  def print_module_run_or_check_usage(command:, description: nil, options: @@module_opts)
+    description ||= command == :check ? 'Check if the target is vulnerable' : "Run the current #{name.downcase} module"
+
+    is_http_mod = mod.is_a?(Msf::Exploit::Remote::HttpClient)
+    is_smb_mod = mod.is_a?(Msf::Exploit::Remote::SMB::Client) || mod.options.include?('SMBUser')
+    is_mysql_mod = mod.is_a?(Msf::Exploit::Remote::MYSQL)
+
+    print_line("Usage: #{command} [options] [RHOSTS]")
+    print_line('')
+    print_line(description)
+    print_line(options.usage)
+    print_line('Examples:')
+    print_line('')
+    print_line("    #{command} 192.168.1.123")
+    print_line("    #{command} 192.168.1.1-192.168.1.254")
+    print_line("    #{command} file:///tmp/rhost_list.txt")
+    print_line("    #{command} http://192.168.1.123/foo") if is_http_mod
+    print_line("    #{command} http://user:pass@192.168.1.123/foo") if is_http_mod
+    print_line("    #{command} HttpTrace=true http://192.168.1.123/foo") if is_http_mod
+    print_line("    #{command} mysql://user:pass@192.168.1.123") if is_mysql_mod
+    print_line("    #{command} SQL='select version()' mysql://user:pass@192.168.1.123") if is_mysql_mod && mod.options.include?('SQL')
+    print_line("    #{command} smb://192.168.1.123") if is_smb_mod
+    print_line("    #{command} smb://user:pass@192.168.1.123") if is_smb_mod
+    print_line("    #{command} LPATH=/tmp/foo.txt smb://user:pass@192.168.1.123/share_name/foo.txt") if is_smb_mod && mod.options.include?('RPATH')
+    print_line('')
+    print_line('Learn more at https://github.com/rapid7/metasploit-framework/wiki/Using-Metasploit')
```

Comment:
TODO: I'll need to update this page with examples of running metasploit modules, or find an existing wiki page to link to / update